src/doc_scan_service/session/retrieve/identity.profile.response.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+const Validation = require('../../../yoti_common/validation');
+const IdentityProfileReportResponse = require('./identity.profile.report.response');
+
+class IdentityProfileResponse {
+  constructor(identityProfile) {
+    Validation.isString(identityProfile.subject_id, 'subject_id', true);
+    this.subjectId = identityProfile.subject_id;
+
+    Validation.isString(identityProfile.result, 'result');
+    this.result = identityProfile.result;
+
+    Validation.isPlainObject(identityProfile.failure_reason, 'failure_reason');
```

Comment:
Aren't `failure_reason` and `identity_profile_report` optional ?

In this corresponding .NET ticket, it's mentioned that these 2 are optional (_and only present in certain cases_).

In any case, not sure if we need much/any of such validation. This data isn't coming from an external client, but from out own Yoti services. So basically such validation tests the behaviour of our Yoti services. So is this ok ?

Just as an example, would it be ok to implement more in-depth validation logic, such as: `failure_reason` = present only if result="ABORDED" ? Such approach might be bad, since we'd duplicate the internal logic performed inside Doc Scan API, and we'd then have to keep it in sync inside the SDKs.

So how much (if any) response validation should we perform inside the SDKs ?

Reply:
> Aren't failure_reason and identity_profile_report optional ?

Yes, they are, in the sense these fields could simply not exist or have nil values (depending on how the session flow happened).

Typically, in the SDK, we are just using a pattern of nil check before in such cases, hence here something like that would do:
```
if (identityProfile.identity_profile_report) { // can add the plain object check as well after 
  this.identityProfileReport = new IdentityProfileReportResponse(...)
}
```

Regarding the validation, as much as it seems a bit useless, we'll keep it consistent to the rest of the SDK code.

As per the `failure_reason`, the .NET has been handling it with an extra class... which is arguable... but for the sake of being aligned, we should do the same, please see the [`FailureReasonResponse`](https://github.com/getyoti/yoti-dotnet-sdk/pull/365/files)... maybe should be named though a bit more specifically, `IdentityProfileFailureReasonResponse` ?

Reply:
Sure so now those properties (failure_reason and identity_profile_report) are optional, and failure_reason now has its own class as suggested.